pkg/ddc/alluxio/shutdown.go

```diff
@@ -170,24 +171,20 @@ func (e *AlluxioEngine) destroyWorkers(workers int32) (err error) {
 		labelExclusiveName = common.Exclusive
 	)
 
-	err = e.List(context.TODO(), nodeList, &client.ListOptions{})
+	labelNames := []string{labelName, labelTotalname, labelDiskName, labelMemoryName, labelCommonName}
+	datasetLabels, err := labels.Parse(fmt.Sprintf("%s=true", labelCommonName))
 	if err != nil {
 		return
 	}
 
-	labelNames := []string{labelName, labelTotalname, labelDiskName, labelMemoryName, labelCommonName}
-
-	runtimeInfo, err := e.getRuntimeInfo()
+	err = e.List(context.TODO(), nodeList, &client.ListOptions{
+		LabelSelector: datasetLabels,
+	})
 	if err != nil {
 		return
 	}
 
-	if runtimeInfo.IsExclusive() {
-		labelNames = append(labelNames, labelExclusiveName)
-	}
-
 	// 1.select the nodes
-	// TODO(cheyang) Need consider node selector
 	var i int32 = 0
 	for _, node := range nodeList.Items {
 		if workers >= 0 {
```

Comment:
Do we still need check i > workers ?

Reply:
if use node selector, we need`t check i> workers.